sympy/stats/stochastic_process_types.py

```diff
@@ -0,0 +1,230 @@
+from sympy import (Symbol, Matrix, MatrixSymbol, S, Indexed, Basic,
+                    Set, And, Tuple, Eq, FiniteSet, ImmutableMatrix)
+from sympy.core.compatibility import string_types
+from sympy.stats.rv import RandomIndexedSymbol, random_symbols, RandomSymbol
+from sympy.stats.symbolic_probability import Probability
+from sympy.stats.stochastic_process import StochasticPSpace
+
+__all__ = [
+    'StochasticProcess',
+    'DiscreteMarkovChain'
+]
+
+class StochasticProcess(Basic):
+    """
+    Base class for all the stochastic process whether
+    discrete or continuous.
+
+    Parameters
+    ==========
+
+    sym: Symbol or string_types
+    state_space: Set
+        The state space of the stochastic process, by default S.Reals.
+        For discrete sets it is zero indexed.
+    """
+
+    is_Continuous = None
+    is_Discrete = None
```

Comment:
The meaning of these is ambiguous here since we can have any combination of discrete/continuous time and discrete/continuous state. I assume this refers to state rather than time

Reply:
> I assume this refers to state rather than time

This was for time. So, should I remove them? Or something like, `is_continuous_time`, `is_discrete_time` will be good to add?

Reply:
Aren't you going for different subclasses?

Reply:
> Aren't you going for different subclasses?

Yeah, when I originally created I was using a common class but now I have created `DiscreteTimeStochasticProcess` and `ContinuousStochasticProcess`(will be created later).
So, I will remove it now. These are quite unnecessary parameters, isn't it? The name is itself suggesting that it is discrete and moreover every stochastic process has their own way of dealing with queries, so they will not be used in computations.

Reply:
sometimes these parameters are used to check the class without using isinstance. Sometimes they can be useful, but not always.